src/lib.rs

```diff
@@ -63,9 +66,18 @@ const PROTOCOL_PERIOD: Duration = Duration::from_millis(1000);
 // TODO: Figure out what an actually optimal size would be.
 const INCOMING_BUFFER_SIZE: usize = 1024;
 
+/// Old recently a peer must have been put into the Known state for us to include them in the list
+/// of peers we send in response to KnownPeersRequest messages.
+const MAX_PEER_SHARE_AGE: Duration = Duration::from_millis(10000);
```

Comment:
I've started keeping track of how old our knowledge about a given peer is, which is updated whenever we receive a message from them. When sharing a list of our known peers, we'll only include the peers that are sufficiently fresh according to this value.

This turns out to be absolutely critical because without it, the mesh will continue passing around ghosts of previously-present nodes indefinitely.